unity/PitayaExample/Assets/Pitaya/PitayaMetrics.cs

```diff
@@ -0,0 +1,172 @@
+            sessionWatch.Start();
+            connectionWatch.Start();
+        }
+
+        public void Update(PitayaNetWorkState state, NetworkError error)
+        {
+            // Session started
+            if (sessionStartedStates.Contains(state))
+            {
+                connectionWatch.Stop();
+                connectionSessionStats.ConnectionTime = connectionWatch.Elapsed;
+            }
+
+            // Session ended
+            if (sessionEndStates.Contains(state) || sessionErroredStates.Contains(state))
+            {
+                stop(state, error);
+            }
+        }
+
+        private void stop(PitayaNetWorkState state=PitayaNetWorkState.Disconnected, NetworkError error=null)
+        {
+            sessionWatch.Stop();
+            connectionSessionStats.SessionDurationSec = sessionWatch.Elapsed;
+            setDisconnectionReason(state, error);
+
+            if (cb != null)
+            {
+                cb(connectionSessionStats);
+            }
+        }
+
+        private void setDisconnectionReason(PitayaNetWorkState state, NetworkError error)
+        {
+            if (state == PitayaNetWorkState.Disconnected)
+            {
+                // Disconnected with errors
+                if (error != null) {
+                    connectionSessionStats.DisconnectionReason = error.Error;
+                    return;
+                }
+
+                // Disconnected normally
+                connectionSessionStats.DisconnectionReason = disconnectionReason.ConnectionEndedNormally;
+                return;
+            }
+
+            // Timeout trying to connect
+            if (state == PitayaNetWorkState.Timeout)
+            {
+                connectionSessionStats.DisconnectionReason = disconnectionReason.ConnectionTimeout;
+                connectionSessionStats.ConnectionFailure = ConnectionFailure.Timeout;
+                if (error != null) {
+                    connectionSessionStats.ConnectionFailureDetails = error.Error;
+                }
+                return;
+            }
+
+            // Disconnected due to errors
+            if (state == PitayaNetWorkState.Error || state == PitayaNetWorkState.FailToConnect)
+            {
+                connectionSessionStats.ConnectionFailure = ConnectionFailure.Error;
+                connectionSessionStats.DisconnectionReason = disconnectionReason.ConnectionErrored;
+
+                if (state == PitayaNetWorkState.FailToConnect)
+                {
+                    connectionSessionStats.DisconnectionReason = disconnectionReason.FailedToConnect;
+                }
+
+                // Error string is set
+                if (error != null) {
+                    connectionSessionStats.ConnectionFailureDetails = error.Error;
+                    return;
+                }
+                
+                return;
+            }
+
+            // Closed connection
+            if (state == PitayaNetWorkState.Closed)
```

Comment:
if you look in `PitayaBinding.cs:16`, the Closed event is never used. I think this could be removed from the code.